doc: Add Rancher Turtles uninstallation notes (#62)
rancher-sandbox/rancher-turtles-docs: docs/getting-started/uninstall_turtles_operator.md

---
sidebar_position: 5
---

# Uninstall Rancher Turtles Operator

This gives an overview of Rancher Turtles Operator uninstallation process.

:::caution

> When installing Rancher Turtles in your Rancher environment, by default, Rancher Turtles enables the Cluster API Operator cleanup. This includes cleaning up Cluster API Operator specific webhooks and deployments that otherwise would block the uninstallation.

Reviewer: ```suggestion
When installing Rancher Turtles in your Rancher environment, by default, Rancher Turtles enables the Cluster API Operator cleanup. This includes cleaning up Cluster API Operator specific webhooks and deployments that otherwise would cause errors in embedded cluster management.
```

> When installing Rancher Turtles in your Rancher environment, by default, Rancher Turtles enables the Cluster API Operator cleanup. This includes cleaning up Cluster API Operator specific webhooks and deployments that otherwise would block the uninstallation.

Reviewer 2: ```suggestion
When installing Rancher Turtles in your Rancher environment, by default, Rancher Turtles enables the Cluster API Operator cleanup. This includes cleaning up Cluster API Operator specific webhooks and deployments that otherwise cause issues with Rancher provisioning.
```

To simplify uninstalling Rancher Turtles, the official Rancher Turtles Helm chart includes a `post-delete` hook that applies these changes, making it transparent to the end user:
- Delete the `mutating-webhook-configuration` and `validating-webhook-configuration` webhooks that are no longer needed.
- Delete the CAPI `deployments` that are no longer needed.
:::

> Once uninstalled, Rancher's `embedded-cluster-api` functionality can be enabled:

Reviewer 2: ```suggestion
Once uninstalled, Rancher's `embedded-cluster-api` feature must be re-enabled:
```

> 1. Create a `feature.yaml` file, with `embedded-cluster-api` set to false:

Reviewer: ```suggestion
1. Create a `feature.yaml` file, with `embedded-cluster-api` set to true:
```

Author: good catch thanks, fixed now
